### PR TITLE
Feature/default matching distance

### PR DESCRIPTION
**Pull request by hakuturu583**
# Description

## Abstract

- Unify the calculation criteria of threshold values used for lane coordinate system pose calculation to some extent for each Entity.
- Changed MiscObject/Pedestrian Entity to take crosswalks into account.

## Background

In #1189 and #1192, it was indicated that the existing lane coordinate system calculations were extremely complex, with different thresholds in various places, etc., and that it was presumed that proper processing was not being done.

## Details

- Unify the calculation criteria of threshold values used for lane coordinate system pose calculation to some extent for each Entity.
  - The matching distance values for the pose calculation in the lane coordinate system varied depending on various conditions. These were simplified and changed to basically use the matching distance determined for each Entity.
  - If the entity is the Ego or Vehicle entity, let $L_m$ be the length of the horizontal bar used in the lane coordinate system calculation and the tread of the front wheels be $t_f$ and the tread of the rear wheels be $t_r$. $$L_m = max(t_r, t_f) + 2.0$$
  - If the entity is the Pedestrian or MiscObject entity, let $L_m$ be the length of the horizontal bar used in the lane coordinate system calculation and width of the bounding box be $L_w$ $$L_m = L_w + 2.0$$

lane_pose_calculation

- Changed MiscObject/Pedestrian Entity to take crosswalks into account
  - They were set not to be taken into account for all Entities at spawn.

### Related issues / Pull requests
#1189 
This pull request is depends on #1192. This pull request become ready for review after merging #1192.

### Regression test result
https://github.com/tier4/sim_evaluation_tools/issues/244

# Destructive Changes

Although this Pull Request does not significantly change the processing flow of the existing algorithm, 100% backward compatibility does not exist because some of the thresholds have been changed.
However, it is assumed that the impact will be limited to a level that does not affect the existing scenario.
In fact, it has been confirmed that regression does not occur in existing scenarios.
See [test log](https://github.com/tier4/sim_evaluation_tools/issues/244) for details.

# Known Limitations

This Pull Request only changes the thresholds of the existing algorithm, and no major processing method changes exist.
Therefore, the specification that the lane coordinate system calculation fails if the horizontal bar extending from the Entity does not touch the curve at the center of the lane remains.

### CODE REVIEW

File: simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp [hunk 1]
```diff
@@ -73,14 +73,17 @@ auto PedestrianActionNode::estimateLaneletPose(const geometry_msgs::msg::Pose &
      * @note Hard coded parameter. 1.0 is a matching threshold for lanelet.
```
Comment (HansRobo) [inline simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp:73]: Invalid comment

File: docs/developer_guide/lane_pose_calculation/UpdateFrame.md [hunk 2]
```diff
@@ -47,11 +47,11 @@ The procedure for calculating the pose in the lane coordinate system at this tim
...
 
-1.[Set the length of the horizontal bar to 2.0 and calculate the pose in the lane coordinate system](https://github.com/tier4/scenario_simulator_v2/blob/5f19d39ef29243396f26225976975f0c 27914c12/simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp#L72-L77)  
-2.[Calculate the pose in the lane coordinate system considering the size of the BoundingBox of the Entity](https://github.com/tier4/scenario_simulator_v2/blob/5f19d39ef29243396f26225976975f0 c27914c12/simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp#L79-L83)  
+1.[Set the length of the horizontal bar to 2.0 and calculate the pose in the lane coordinate system](https://github.com/tier4/scenario_simulator_v2/blob/5f19d39ef29243396f26225976975f0c27914c12/simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp#L72-L77)  
+2.[Calculate the pose in the lane coordinate system considering the size of the BoundingBox of the Entity](https://github.com/tier4/scenario_simulator_v2/blob/5f19d39ef29243396f26225976975f0c27914c12/simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp#L79-L83)
```
Comment (HansRobo) [inline docs/developer_guide/lane_pose_calculation/UpdateFrame.md:51]: The links and descriptions are old ones.
Please match the documentation with the latest code.

File: simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp [hunk 3]
```diff
@@ -520,7 +520,24 @@ class EntityManager
       } else {
         entity_status.pose = pose;
 
-        if (const auto lanelet_pose = toLaneletPose(pose, parameters.bounding_box, false);
+        /// @note If the entity is pedestrian or mis object, we have to consider matching to crosswalk lanelet.
```
Comment (HansRobo) [inline simulation/traffic_simulator/include/traffic_simulator/entity/entity_manager.hpp:523]:
```suggestion
        /// @note If the entity is pedestrian or misc object, we have to consider matching to crosswalk lanelet.
```

File: simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp (hunk 1 above)
Comment (hakuturu583) [reply simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp]: Thanks, I fixed.

Comment (HansRobo) [reply simulation/behavior_tree_plugin/src/pedestrian/pedestrian_action_node.cpp]: I confirmed in https://github.com/tier4/scenario_simulator_v2/pull/1194/commits/d0bfdf62a14f6e2da2a78259ccffbd42c840d375

File: docs/developer_guide/lane_pose_calculation/UpdateFrame.md (hunk 2 above)
Comment (hakuturu583) [reply docs/developer_guide/lane_pose_calculation/UpdateFrame.md]: Fixed in https://github.com/tier4/scenario_simulator_v2/pull/1194/commits/17b5d94e0c85728d58287556808cff182959389e and https://github.com/tier4/scenario_simulator_v2/pull/1194/commits/e02b39cdd5d58dedbea8788c8d42154cf1dac2c5

**Review by HansRobo — APPROVED** [review]

**Pull request merged**